src/VirtualClient/VirtualClient.Dependencies/Packaging/GitRepoClone.cs

```diff
@@ -47,13 +47,30 @@ public Uri RepoUri
             }
         }
 
+        /// <summary>
+        /// Git checkout
+        /// </summary>
+        public string Checkout
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(GitRepoClone.Checkout), string.Empty);
+            }
+
+            set
+            {
+                this.Parameters[nameof(GitRepoClone.Checkout)] = value;
+            }
+        }
+
         /// <summary>
         /// Executes the git clone operation.
         /// </summary>
         protected override async Task ExecuteAsync(EventContext telemetryContext, CancellationToken cancellationToken)
         {
             telemetryContext.AddContext("repoUri", this.RepoUri);
             telemetryContext.AddContext("packagesDirectory", this.PlatformSpecifics.PackagesDirectory);
+            telemetryContext.AddContext("checkout", this.Checkout);
```

Comment:
Let's use the actual terms that are related to Git for this feature. You can expose both a "Commit" property as well as a "Branch" property.

This makes it more clear to a user/reader of a profile.

Reply:
so "Commit" but you could use branch, or commit or tag

Reply:
Good points. Let's just stay with "Checkout" so long as we have good documentation describing the possible uses. I retract my statements above.